llvm/lib/CodeGen/SelectionDAG/LegalizeIntegerTypes.cpp

```diff
@@ -4574,20 +4584,29 @@ void DAGTypeLegalizer::ExpandIntRes_ShiftThroughStack(SDNode *N, SDValue &Lo,
     Init = DAG.getNode(ISD::BUILD_PAIR, dl, StackSlotVT, AllZeros, Shiftee);
   }
   // And spill it into the stack slot.
-  Ch = DAG.getStore(Ch, dl, Init, StackPtr, StackPtrInfo, StackSlotAlignment);
+  Ch = DAG.getStore(Ch, dl, Init, StackPtr, StackPtrInfo);
```

Comment:
This store is not getting the correct alignment set. The stack alignment should be the next argument if using the MachinePointerInfo overloads

Reply:
Alignment is calculated by getEVTAlign inside the function, which gives the correct alignment in this case or? Or do you mean that I should I add StackAlign explicitly?

Reply:
Magically figuring it out inside this is more confusing than just explicitly passing it, especially since you already have alignment logic here